File: 1603-design-parking-system/1603-design-parking-system.cs
public class ParkingSystem
{

  private int Big { get; set; }
  private int Medium { get; set; }
  private int Small { get; set; }

  public ParkingSystem(int big, int medium, int small)
  {
    Big = big;
    Medium = medium;
    Small = small;
  }


  public bool AddCar(int carType)
  {
    // small
    if (carType == 3 && Small - 1 >= 0)
    {
      Small -= 1;
      return true;
    }
    // medium
    if (carType == 2 && Medium - 1 >= 0)
    {
      Medium -= 1;
      return true;
    }
    // big
    if (carType == 1 && Big - 1 >= 0)
    {
      Big -= 1;
      return true;
    }
    return false;
  }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj.AddCar(carType);
 */

File: 1603-design-parking-system/1603-design-parking-system.go
type ParkingSystem struct {
	bigMaxCapa     int
	bigCurrCapa    int
	mediumMaxCapa  int
	mediumCurrCapa int
	smallMaxCapa   int
	smallCurrCapa  int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{bigMaxCapa: big, mediumMaxCapa: medium, smallMaxCapa: small}
}

func (this *ParkingSystem) AddCar(carType int) bool {
	isValid := false

	switch carType {
	case 1:
		if this.bigCurrCapa < this.bigMaxCapa {
			this.bigCurrCapa++
			isValid = true
		}
	case 2:
		if this.mediumCurrCapa < this.mediumMaxCapa {
			this.mediumCurrCapa++
			isValid = true
		}
	case 3:
		if this.smallCurrCapa < this.smallMaxCapa {
			this.smallCurrCapa++
			isValid = true
		}
	}

	return isValid
}


/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */